### PR TITLE
Add support for Zhaoxin's x86 platfrom

### PR DESCRIPTION
**Pull request by mormenthan**
VIA and Shanghai United Investment Co.,Ltd. found Shanghai ZhaoXin, which is a fabless x86 CPU IC design company. ZhaoXin has issued ZX-C, ZX-D x86 processors, which have 'Shanghai' CPU vendor id.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by mspncp** [comment]
Hello @mormenthan, I am not involved in reviewing this PR, just wanted to let you know that the word "platfrom" in the PR title and the commit message is misspelled, it should be "platform" instead.

File: engines/asm/e_padlock-x86_64.pl [hunk 1]
```diff
@@ -57,11 +57,20 @@
 	cpuid
 	xor	%eax,%eax
 	cmp	\$`"0x".unpack("H*",'tneC')`,%ebx
-	jne	.Lnoluck
+	jne	.Lshanghai
 	cmp	\$`"0x".unpack("H*",'Hrua')`,%edx
 	jne	.Lnoluck
 	cmp	\$`"0x".unpack("H*",'slua')`,%ecx
 	jne	.Lnoluck
+	jmp .LshanghaiEnd
```
Comment (dot-asm) [inline engines/asm/e_padlock-x86_64.pl]: For more consistent look replace space after jmp with tab.

**Comment by dot-asm** [comment]
As for typo in "platfrom" (thanks, Matthias), and rather in commit message than subject for this request. Even though committer can fix it upon commit, it might be forgotten. So I'd suggest to fix typo in commit message. Just `git commit --amend` and `git push -f`.

**Comment by mormenthan** [comment]
@mspncp
Thxs for your tips.

**Comment by mormenthan** [comment]
@dot-asm 
1. change shanghai with zhaoxin typo
2. jmp with tab

**Comment by mspncp** [comment]
And don't forget the "platfrom" ;-)

**Comment by dot-asm** [comment]
I'll be squashing, so I undertake remembering to fix it.

**Comment by mattcaswell** [comment]
@dot-asm...will you merge? Preferably before the code freeze later today... ;-)

**Comment by dot-asm** [comment]
> will you merge?

Sure...

**Comment by dot-asm** [comment]
Merged. Thanks!

**Comment by richsalz** [comment]
Why did we do this, and not make it an external engine?

**Comment by t-j-h** [comment]
Because it is useful. And we don't have a policy that says we make these external. 
I too would have approved it.
They applied Andy's feedback on approach and on the PR itself.

**Comment by dot-asm** [comment]
> Why did we do this, and not make it an external engine?

My rationale *here* is that it doesn't actually change current *functionality*, just makes it possible to use *existing* code on yet another processor. When/if/as additional functionality is added, I'd too argue that it should become external project. And at that point suggestion would also be to remove e_padlock code from our tree. But till it becomes usable, I'd argue that it's appropriate to facilitate existing code re-use.

**Pull request closed without merging**